main/views.py

```diff
@@ -26,16 +27,33 @@ class HomeView(TemplateView):
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        downtowns = Downtown.objects.all()
+        query = self.request.GET.get('query', '')
+        downtowns = Downtown.objects.filter(name__contains=query)
+        users = User.objects.filter(
```

Comment:
Use Django's get_user_model instead of directly referring to the user model. That the code works even if the user class changes.